### ISSUE TITLE
🐛 Bug Report: Generated app fails to build when reserved names are used for an entity name

### ISSUE THREAD

**Issue by yuval-hazaz** [amplication/amplication#2269]
### What happened?

When a user creates an entity with a reserved name, the generated app failed to build.

For example, when the user uses the name "Error" for an entity, the following exception is thrown at Cloud Build when building the docker image for the sandbox

```
Step #1: Step #2: /app/server/node_modules/ts-node/src/index.ts:513
Step #1: Step #2:     return new TSError(diagnosticText, diagnosticCodes)
Step #1: Step #2:            ^
Step #1: Step #2: TSError: ⨯ Unable to compile TypeScript:
Step #1: Step #2: scripts/seed.ts(14,21): error TS2554: Expected 0 arguments, but got 1.
```

The same error can see seen when running the seed script locally

![image](https://user-images.githubusercontent.com/43705455/152522549-dd96031d-f07d-449c-9bf9-02117ff7d1b1.png)

### What you expected to happen

The user should not be able to create entities (and entity fields) with names that are reserved

### How to reproduce

Create an entity with the name Error and commit the changes to deploy the code to sandbox - the deployment will fail

### Amplication version

0.11.2

**Comment by noyjs** [comment]
Hey, can I do it?

**Comment by yuval-hazaz** [comment]
sure @noyjs , I assigned it to you

**Comment by yuval-hazaz** [comment]
for this issue, we need to create a list of reserved names on the server and use it as part of the validation before creating or updating an entity and entity field

**Comment by yuval-hazaz** [comment]
also, the list may initially include all these word

https://github.com/microsoft/TypeScript/issues/2536

**Comment by noyjs** [comment]
should be fixed by #2272

**Comment by mshidlov** [comment]
this issue was merged to release `0.11.3`
